### ISSUE TITLE
Linux + File Regex true returns NullPointerException

### ISSUE THREAD

**Issue by gabrielprioli** [dynaTrace/Dynatrace-Log-File-Monitor#5]
When executing the plugin with the "File Regex" set to true on a Linux machine i got a NullPointerException.

```
2017-01-23 22:40:22 INFO [com.logfile.WP@] SSH Publickey authentication
2017-01-23 22:40:22 INFO [com.logfile.WP@] Authenticated with Public Key
2017-01-23 22:40:22 INFO [com.logfile.WP@] Authentication Successful
2017-01-23 22:40:22 INFO [com.logfile.WP@] Connection setup successful
2017-01-23 22:40:22 INFO [com.logfile.WP@] Sending command string: uname
2017-01-23 22:40:22 INFO [com.logfile.WP@] Received result to parse: Linux
2017-01-23 22:40:22 INFO [com.logfile.WP@] Connecting to /home/vagrant/mylog_* on 172.16.13.115...
2017-01-23 22:40:22 INFO [com.logfile.WP@] Sending command string: ls /home/vagrant/mylog_* -ltc | awk '{print $9}'
2017-01-23 22:40:22 WARNING [com.logfile.WP@] Cannot establish a connection...
2017-01-23 22:40:22 WARNING [com.logfile.WP@] java.lang.NullPointerException
	at com.dynatrace.diagnostics.plugins.logfile.connection.SSHConnectionMethod.executeStringCommand(SSHConnectionMethod.java:55)
```
Looking at the code I see that [line 232 of WP.java](https://github.com/Dynatrace/Dynatrace-Log-File-Monitor/blob/master/src/com/logfile/WP.java#L232) calls [getConnectionMethod()](https://github.com/Dynatrace/Dynatrace-Log-File-Monitor/blob/bc6c6c/src/com/dynatrace/diagnostics/plugins/logfile/connection/ConnectionMethod.java#L74) which simply returns a new SSHConnectionMethod object.

```
String filereturn = ConnectionMethod.getConnectionMethod(sshConnMethod).executeCommand(findcommand, "");

```
```
public static ConnectionMethod getConnectionMethod(String method) {
		if (method.equals("LOCAL")) {
			return new LocalConnectionMethod();
		} else {
			return new SSHConnectionMethod();
		}
	}
```
After getting the new `ConnectionMethod`, the code calls `executeCommand` which calls `executeStringCommand` but this `ConnectionMethod` object is empty and fails when it tries to use the `conn` field.

**Comment by gabrielprioli** [comment]
I've added a change that fixes this by reusing the connMethod object which was already set up.

[https://github.com/gabrielprioli/Dynatrace-Log-File-Monitor/commit/b804f12c832a39d84cf2b82e17b4a4f1fcf82226
](https://github.com/gabrielprioli/Dynatrace-Log-File-Monitor/commit/b804f12c832a39d84cf2b82e17b4a4f1fcf82226
)

Please take a look and I'll make any modification if needed. Otherwise let me know and I'll pull request to this repo. =)

**Comment by Dynatrace-Jeroen-Hautekeete** [comment]
Fix is included in release 3.16.4
